test/common/http/header_utility_test.cc

```diff
@@ -587,7 +590,7 @@ invert_match: true
   EXPECT_TRUE(HeaderUtility::matchHeaders(unmatching_headers, header_data));
 }
 
-TEST(MatchHeadersTest, HeaderPrefixMatch) {
+TEST(MatchHeadersTest, DEPRECATED_FEATURE_TEST(HeaderPrefixMatch)) {
```

Comment:
Do we have sufficient coverage such that when these are removed, the non-`DEPRECATED_FEATURE_TEST` fully cover the alternative `string_match` behaviors?

Reply:
we should have sufficient coverage, my last commit only added `DEPRECATED_FEATURE_TEST` to the unit tests that are **directly** testing the deprecated fields (i.e. `exact_match`, `prefix_match`, etc). I also added the test for the `string_match` field.

That's also why I didn't add the tag to `MayMatchOneOrMoreRequestHeader` but instead changed the test to use `string_match` field because it's testing more than the deprecated fields.

Reply:
Makes sense, thanks.